proto-public/pbmesh/v2beta1/pbproxystate/protocol.proto

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+syntax = "proto3";
+
+package hashicorp.consul.mesh.v2beta1.pbproxystate;
+
+// +kubebuilder:validation:Enum=PROTOCOL_UNSPECIFIED;PROTOCOL_TCP;PROTOCOL_HTTP;PROTOCOL_HTTP2;PROTOCOL_GRPC
+// +kubebuilder:validation:Type=string
+enum Protocol {
+  PROTOCOL_UNSPECIFIED = 0;
+  PROTOCOL_TCP = 1;
+  PROTOCOL_HTTP = 2;
+  PROTOCOL_HTTP2 = 3;
+  PROTOCOL_GRPC = 4;
+
+  // Protocol Mesh indicates that this port can speak Consul's mTLS based mesh protocol.
+  PROTOCOL_MESH = 5;
```

Comment:
I guess we don't need it for proxy state, assuming we never try to "convert backwards" in tests or something like that.

If we omit it, it'll make casting between the catalog and proxystate versions of this enum impossible if/when other protocols are added in the future.

Two options I can see:
1. Punt for now, comment that it's unused and kept for parity, and maybe validate it's never used on the proxystate side
2. Stop casting between the two enums and convert between them more explicitly

I'm fine w/ either. @jmurret @ishustava if you have a preference happy to go w/ that, otherwise I might go w/ 2. since it's less complicated from a behavior standpoint

Reply:
You can leave it, and just reject it in the validate function for PST.

Reply:
It's not a valid protocol for an envoy cluster though, and so it feels like an invalid option in proxy state as well. Maybe adding validation to prevent someone from setting it would be good?

I see that right now we ignore protocols that are not l7 for protocol options in envoy generation code. I think rejecting mesh protocol explicitly would prevent us from making mistakes in the sidecar proxy controller potentially where you can set cluster protocol incorrectly. That would be nice to add but doesn't have to block this PR.

Reply:
Agreed, thanks y'all. Added explicit rejection to the PST validator and a test to cover; I see the `kubebuilder:validation` annotations also exclude it already.